packages/xo-server/src/api/sr.js

```diff
@@ -361,6 +398,41 @@ createExt.resolve = {
   host: ['host', 'host', 'administrate'],
 }
 
+// -------------------------------------------------------------------
+// This function helps to detect all ZFS pools
+// Return a dict of pools with their parameters { <poolname>: {<paramdict>}}
+
+export async function probeZfs({ host }) {
+  const xapi = this.getXapi(host)
+  try {
+    const result = await xapi.call(
+      'host.call_plugin',
+      host._xapiRef,
+      'zfs.py',
+      'list_zfs_pools',
+      {}
+    )
+    return JSON.parse(result)
+  } catch (error) {
+    if (
+      error.code === 'XENAPI_MISSING_PLUGIN' ||
+      error.code === 'UNKNOWN_XENAPI_PLUGIN_FUNCTION'
+    ) {
+      return {}
+    } else {
+      throw error
+    }
+  }
+}
```

Comment:
Could you add an example of the result value to help with the UI implementation?